LoopKitUI/Views/Settings Editors/TherapySettingsView.swift

```diff
@@ -322,6 +327,55 @@ extension TherapySettingsView {
     }
 }
 
+// MARK: Navigation
+
+extension TherapySettingsView {
+
+    func screen(for setting: TherapySetting) -> (_ dismiss: @escaping () -> Void) -> AnyView {
+        switch setting {
+        case .suspendThreshold:
+            return { dismiss in
+                AnyView(SuspendThresholdEditor(mode: mode, therapySettingsViewModel: viewModel, didSave: dismiss).environment(\.dismiss, dismiss))
+            }
```

Comment:
will `@ViewBuilder` help remove the `AnyView` here?

Reply:
Unfortunately, no, I don't think so. If the `screen` function simply returned a view then we could change the return value to `some View`, add `@ViewBuilder`, and ditch the `AnyView`. However, this function returns **a closure that returns a view** and that closure must have a concrete type AFAIK. Thus, `AnyView`.

BTW, I mostly just copied this existing code over from `TherapySettingsViewModel` with only small changes to add the `mode` parameter to each view initializer.